segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataCleaner.java

```diff
@@ -179,6 +180,26 @@ protected void doStop() {
                 }, this.executor);
     }
 
+    private CompletableFuture<Void> deleteUnusedTransientSegments() {
+        long lastSeqNo = this.lastIterationSequenceNumber.getAndSet(this.metadata.getOperationSequenceNumber());
```

Comment:
This has side effects. It's also invoked in `evictUnusedSegments` which means that likely the `MetadataCleaner` will not function correctly anymore. 

I think we may have to factor some of this code out into `runInternal` and then pass whatever we need as arguments to both these methods.